### PR TITLE
Jdbc 50 add test suite prep statement param types

### PR DESCRIPTION
**Pull request by rpastrana**

### CODE REVIEW

**Comment by rpastrana** [comment]
@garonsky please review 2ff8faa

**Comment by rpastrana** [comment]
@garonsky please look over new changes.

File: src/main/java/org/hpccsystems/jdbcdriver/HPCCJDBCUtils.java [hunk 1]
```diff
@@ -729,6 +734,77 @@ public static int convertECLtypeCode2SQLtype(EclTypes ecltype)
...
+            return Float.parseFloat( value );
+        else if( Double.TYPE == clazz )
+            return Double.parseDouble( value );
+        else if ( Short.TYPE == clazz)
+            return Short.parseShort(value);
+        else if (Character.TYPE == clazz)
+            return value.charAt(0);
+        else if (java.math.BigDecimal.class == clazz)
+            return new BigDecimal(value.replaceAll(",", ""));
+        else if (java.sql.Date.class == clazz) //ASSUMES yyyy-MM-dd!!!
```
Comment (garonsky) [inline src/main/java/org/hpccsystems/jdbcdriver/HPCCJDBCUtils.java]: Catch illegal argument exception?

File: src/main/java/org/hpccsystems/jdbcdriver/HPCCPreparedStatement.java [hunk 2]
```diff
@@ -123,6 +134,63 @@ public ResultSet executeQuery() throws SQLException
         return result;
     }
 
+    public static String primitiveToString(Object x) throws IOException
```
Comment (garonsky) [inline src/main/java/org/hpccsystems/jdbcdriver/HPCCPreparedStatement.java]: double check this function throws

File: src/test/java/org/hpccsystems/jdbcdriver/tests/HPCCDriverTest.java [hunk 3]
```diff
@@ -280,8 +280,19 @@ private void executeFreeHandSQL(String sql, boolean expectPass, int minResults,
...
-                    p.setObject(i + 1, csvlines[i]);
-                    prepParamValue = csvlines[i];
+                    String value = null;
+                    String currline = csvlines[i];
+                    String [] valuesAndTheirTypes = currline.split(",");
+                    if (valuesAndTheirTypes.length > 0)
+                    {
+                        value = String.valueOf(valuesAndTheirTypes[0]);
+                        if (valuesAndTheirTypes.length > 1)
+                            p.setObject(i + 1, HPCCJDBCUtils.deserializeSQLTypesToJava(valuesAndTheirTypes[1].toLowerCase(), value));
```
Comment (garonsky) [inline src/test/java/org/hpccsystems/jdbcdriver/tests/HPCCDriverTest.java]: why make them lowercase when you make them uppercase in the function?

File: src/main/java/org/hpccsystems/jdbcdriver/HPCCPreparedStatement.java [hunk 4]
```diff
@@ -123,6 +134,63 @@ public ResultSet executeQuery() throws SQLException
...
+            else if (x instanceof Time)
+                return ((Time)x).toString();
+            else if (x instanceof java.sql.Date)
+                return ((java.sql.Date)x).toString();
+            else if (x instanceof Timestamp)
+                return ((Timestamp)x).toString();
+            else if (x instanceof InputStream)
+            {
+                ByteArrayOutputStream result = new ByteArrayOutputStream();
+                byte[] buffer = new byte[1024];
```
Comment (garonsky) [inline src/main/java/org/hpccsystems/jdbcdriver/HPCCPreparedStatement.java]: consider larger buffer

File: src/main/java/org/hpccsystems/jdbcdriver/HPCCJDBCUtils.java (hunk 1 above)
Comment (garonsky) [inline src/main/java/org/hpccsystems/jdbcdriver/HPCCJDBCUtils.java]: Catch illegal argument exception?

Comment (rpastrana) [inline src/main/java/org/hpccsystems/jdbcdriver/HPCCJDBCUtils.java]: This function throws, it's handled by the caller

**Pull request merged**